lib/internal/bootstrap_node.js

```diff
@@ -268,15 +268,24 @@
     var console;
     Object.defineProperty(global, 'console', {
       configurable: true,
-      enumerable: true,
+      enumerable: false,
       get: function() {
         if (!console) {
           console = NativeModule.require('console');
           installInspectorConsoleIfNeeded(console,
                                           inspectorConsole,
                                           wrapConsoleCall);
         }
+        Object.defineProperty(global, 'console', {
+          configurable: true,
+          writable: true,
+          enumerable: false,
+          value: console
+        });
         return console;
+      },
+      set: function(customConsole) {
+        console = customConsole;
```

Comment:
Suggested solution. Good point actually. I added a separate test suite specifically for the problem. It sets global.console = undefined above requires. Otherwise something in requires read global.console.